src/mavlink-router/mainloop.cpp

```diff
@@ -542,7 +542,7 @@ Timeout *Mainloop::add_timeout(uint32_t timeout_msec, std::function<bool(void*)>
 
     assert_or_return(t, NULL);
 
-    t->fd = timerfd_create(CLOCK_MONOTONIC, 0);
+    t->fd = timerfd_create(CLOCK_MONOTONIC, TFD_NONBLOCK);
```

Comment:
yes, all fds should be in NONBLOCK mode, but we should also not trying to read a disarmed timeout.